server/server.go

```diff
@@ -81,16 +115,22 @@ func (s *Server) Serve() error {
 		UseAuth:            s.useAuth(),
 	}, service)
 
+	// Add chronograf's version header to all requests
 	s.handler = Version(s.BuildInfo.Version, s.handler)
 
-	var err error
-	s.Listener, err = net.Listen("tcp", net.JoinHostPort(s.Host, strconv.Itoa(s.Port)))
+	if s.useTLS() {
+		// Add HSTS to instruct all browsers to change from http to https
+		s.handler = HSTS(s.handler)
```

Comment:
So, HSTS really informs the client to cache that HTTPS should be used for this domain and potentially its subdomains for a length of time.

If that client that has the value cached ever receives HTTP instead it would know to reject it as a possible man-in-the-middle.  

So for normal API calls to HTTPS from, say, curl, it would not use HSTS as that value is probably not cached.  

HSTS doesn't redirect HTTP to HTTPS but rather is only used on HTTPS responses.